pkgs/development/tools/misc/gdb/default.nix

```diff
@@ -16,21 +16,19 @@
   ]
 }:
 
-let
-  basename = "gdb-${version}";
-  version = "9.2";
-in
-
 assert pythonSupport -> python3 != null;
 
 stdenv.mkDerivation rec {
+  pname = "gdb";
```

Comment:
Wouldn't this break `name`/`pname` consistency?

Reply:
Mmmh, from my understanding, the derivation of `name` from `pname` and `version` is just a default behavior (to avoid repetition across packages) as proposed by NixOS/rfcs#35 and implemented in #49398; I don't interpret this as a hard requirement (I might be wrong).

I am not too sure **why** `gdb.name` is what it is now... @pierron, @globin, @lsix?

Reply:
do you mind expanding on what you mean by consistency?

The refactor looks like it retains the original behavior for derivation name. (I didn't want to actually build it, so I just pulled outpath from .drv)

```
[14:48:58] jon@jon-desktop /home/jon/projects/nixpkgs (master)
$ nix-build -A gdb
these paths will be fetched (6.53 MiB download, 34.07 MiB unpacked):
...
/nix/store/kk517cqri65mpy3bjwhjz0kg8csn52a4-gdb-9.2
[14:55:43] jon@nixos ~/projects/nixpkgs (gdb)
$ nix show-derivation /nix/store/ywpwqspy4c439819yblx5850g8zdscz2-gdb-9.2.drv
{
  "/nix/store/ywpwqspy4c439819yblx5850g8zdscz2-gdb-9.2.drv": {
    "outputs": {
      "out": {
        "path": "/nix/store/3777ik4rss5vbzqyfsaxpmdm9zr21aq7-gdb-9.2"
      }
    },
```

Reply:
@jonringer I would expect the following invariants: `name == "${pname}-${version}"` and `(builtins.parseDrvName name).name == pname`, but this is not always the case for resulting expression.

@Pamplemousse This may be not a hard requirement, but I think it is a good idea to keep surprises to minimum. I suppose we could drop `name` and keep just the `pname` like we did for many other expressions.

Reply:
I'm assuming the name construction (which pre dates this pr) would break the pname assertion

Reply:
Which `pname` assertion?
Should I remove the `name` attribute altogether then?

Reply:
Make it just like GDB: 

https://github.com/NixOS/nixpkgs/blob/7d6e7b3cd3573236a24a3ff39a3a04bcc1a4c9b6/pkgs/development/tools/misc/binutils/default.nix#L25-L45

i.e. we have
```
basename = "gdb";
```
```
pname = targetPrefix + basename;
```

Reply:
His `(builtins.parseDrvName name).name == pname` would fail in the case of cross compiling

however, it would fail from existing behavior as the file currently has:
```
  name =
    stdenv.lib.optionalString (stdenv.targetPlatform != stdenv.hostPlatform)
                              (stdenv.targetPlatform.config + "-")
    + basename;
```

to make it not fail, you would need to remove the above code for setting `gdb`

Reply:
Yeah if it's like GDB, with no explicit `name = ...`, then we don't have either of those problems.